ffi/firewood_test.go

```diff
@@ -168,9 +172,17 @@ func TestInsert100(t *testing.T) {
 
 			hash, err := db.Root()
 			assert.NoError(t, err, "%T.Root()", db)
-			assert.Lenf(t, hash, 32, "%T.Root()", db)
-			// we know the hash starts with 0xf8
-			assert.Equalf(t, byte(0xf8), hash[0], "First byte of %T.Root()", db)
+
+			// Assert the hash is exactly as expected. Test failure indicates a
+			// non-hash compatible change has been made since the string was set.
+			// If that's expected, update the string to fix the test.
+			expectedHashHex := "f858b51ada79c4abeb6566ef1204a453030dba1cca3526d174e2cb3ce2aadc57"
+			if os.Getenv("TEST_FIREWOOD_ETHHASH") == "true" {
+				expectedHashHex = "c25a0076e0337d7c982c3c9dfa445c8088242a0a607f9d9def3762765bcb0fde"
+			}
```

Comment:
Inlining seems simplest, can move to improved test vectors as we improve the tests

Reply:
Can we name these constants somewhere? Having unnamed magics is confusing.

Reply:
Updated